secret-provider/src/main/scala/io/lenses/connect/secrets/cache/ValueWithTtl.scala

```diff
@@ -0,0 +1,39 @@
+package io.lenses.connect.secrets.cache
+
+import java.time.Clock
+import java.time.OffsetDateTime
+import java.time.temporal.ChronoUnit
+import scala.concurrent.duration.Duration
+import scala.concurrent.duration.MILLISECONDS
+
+case class Ttl(originalTtl: Duration, expiry: OffsetDateTime) {
+
+  def ttlRemaining(implicit clock: Clock): Duration =
+    Duration(expiry.toInstant.toEpochMilli - clock.millis(), MILLISECONDS)
+
+  def isAlive(implicit clock: Clock) = expiry.toInstant.isAfter(clock.instant())
+
+}
+
+object Ttl {
+  def apply(ttl: Option[Duration], defaultTtl: Option[Duration])(implicit clock: Clock): Option[Ttl] = {
+    val applicableTtl = Seq(ttl, defaultTtl).flatten.filterNot(_.toMillis == 0).headOption
```

Comment:
The code might read better and also is more performant (Not that in this case it matters)
```
ttl.orElse(defaultTtl). map(final => Ttl(final, ttlToExpiry(final))
```

Is there a scenario where defaultTtl is None? Normally a default should actually provide a value.

Reply:
Is there a need for an infinite expiry?

Reply:
> Is there a need for an infinite expiry?

Yes, if no default ttl is specified, and the secret in Vault does not have a ttl/lease time, then there will be no expiry.